Updated docs/install.rst. Fixes issue #894

Ritvik-Sapra opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add it in the CHANGELOG file under Unreleased section.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary
The file "install.rst" in docs explained creating and activating a anaconda environment. It showed the following command.
`source activate name_of_my_env`
However, for new anaconda version (>4.6), this command is depreciated and instead following command is used.
`conda activate name_of_my_env`

This has been updated in the file.

### Details and comments
Working on qiskit using anaconda is very important. More than virtual environments, anaconda is preferred by many people. `source activate` or `activate` is a command for older versions of anaconda. New versions are depreciating this command and for following versions, this command might not even be recognized. Hence, _this pull request updates the docs_ by giving instructions for new version.
While installing, anaconda still gets installed with older versions and many people may not update it. Therefore, the _older command is not deleted_. Just new command is added for current and future version of anacoda.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Qiskit/qiskit?pullRequest=957) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Qiskit/qiskit?pullRequest=957) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Qiskit/qiskit?pullRequest=957) it.</sub>

SooluThomas commented on docs/install.rst:
```diff
@@ -39,6 +39,11 @@ Create a minimal environment with only Python installed in it.
 
   conda create -n name_of_my_env python=3
 
+If you are using conda version 4.6 or above, which is recommended, use following command to activate your environment:
+
+  conda activate name_of_my_env
+This command can be used irrespective of Windows or Unix/Linux. Simply type these commands in anaconda prompt.
+
```
I think these lines should be added after 
```rst
.. code:: sh

   activate name_of_me_env
```
or put the older commands (used for conda versions prior to 4.6) in a note. 
```rst
.. note::
   
   If you are using conda versions prior to 4.6, use:
   
   For Linux or macOS: ``source activate name_of_my_env``
   For Windows: ``activate name_of_my_env``
```

And add Windows instructions after/in L33-L34. Maybe something like:
```rst
If you are using Linux or macOS, open a terminal window in the directory where you want to work. 
For windows, open Anaconda Prompt.
```

Also, to fix the lint error, each line should not be more than 100 characters.

Ritvik-Sapra replied:
Yes, I actually did this before. But then updated the commit and put it before because mostly new versions will be used. So any user will feel easier to find these instructions. However, your suggested way makes it even more clean. 
I updated the required changes.

Ritvik-Sapra commented:
![image](https://user-images.githubusercontent.com/54613995/85983378-0b92e300-ba05-11ea-9a70-a1563c94c9e9.png)
I am getting this error when i'm trying to run ``tox -elint``. 

I first installed tox using ``pip install tox`` with cmd. and I installed pycodestyle. But its still showing this error.

SooluThomas commented:
I don't think the change made in the commit https://github.com/Qiskit/qiskit/pull/957/commits/bf329ee19c9a5b16128ce681b35f6a237dd65c00 is necessary. Please revert it and you can also check here - https://travis-ci.com/github/Qiskit/qiskit/jobs/355186912#L470 to see the lines which are failing lint tests.

Ritvik-Sapra commented:
Hey, yes. I will revert back to jupyter executive. @nonhermitian  Actually that line kept showing as an error in VS code. So I tried installing Sphinx and then there were more errors. I will revert back all the changes and then run the code again in a conda env now as @SooluThomas guided..

Ritvik-Sapra commented:
![image](https://user-images.githubusercontent.com/54613995/86156676-f660a600-bb23-11ea-9e27-3ca628d8dcf0.png)
I reverted back jupyter-executive and removed whitespaces from 36 and 37 as told in [this link](https://travis-ci.com/github/Qiskit/qiskit/jobs/355186912#L470) @SooluThomas 
But this error shown above is being given by VS code. This is why I removed in the first place. Any idea why? 😐

SooluThomas commented on docs/install.rst:
```diff
@@ -96,7 +94,7 @@ versions of macOS) you'll need to put ``qiskit[visualization]`` in quotes:
   them into your environment with Python to begin working.
 
-.. code:: python
+.. jupyter-executive:: python
```
```suggestion
.. jupyter-execute::
```

SooluThomas commented on docs/install.rst:
```diff
@@ -162,7 +160,7 @@ returns only the version for the ``qiskit-terra`` package. This is because
 instead is part of the ``qiskit-terra`` package.
 
-.. jupyter-execute::
+.. jupyter-executive::
```
```suggestion
.. jupyter-execute::
```

Ritvik-Sapra commented:
Done. But that VS code is still showing error in this line.

SooluThomas commented:
I'm not sure why your VSCode is showing the error. But I think this PR is good to be merged.

SooluThomas commented on docs/install.rst:
```diff
@@ -96,7 +94,7 @@ versions of macOS) you'll need to put ``qiskit[visualization]`` in quotes:
   them into your environment with Python to begin working.
 
-.. code:: python
+.. jupyter-execute::
```
```suggestion
.. code:: python
```
